scripts/manage-db.py

```diff
@@ -45,6 +45,20 @@ def cleanup_releases(trans, nightly_age, dryrun=True):
         trans.execute("DELETE releases FROM releases" + query)
 
 
+def cleanup_releases_json(trans, nightly_age, dryrun=True):
+    query = RELEASES_CLEANUP_CONDITION.format(nightly_age=nightly_age)
```

Comment:
suggestion (non-blocking)

We could use `text` and let sqlalchemy handle the parameter substitution

```diff
diff --git a/scripts/manage-db.py b/scripts/manage-db.py
index 4306c6c6..10528d52 100755
--- a/scripts/manage-db.py
+++ b/scripts/manage-db.py
@@ -7,6 +7,7 @@ from os import path
 from subprocess import run
 
 from sqlalchemy.engine.url import make_url
+from sqlalchemy import text
 
 from auslib.blobs.base import createBlob  # noqa: E402
 from auslib.db import AUSDatabase  # noqa: E402
@@ -24,7 +25,7 @@ LEFT JOIN rules rules_mapping ON (name=rules_mapping.mapping)
 WHERE name LIKE '%%nightly%%'
 AND name NOT LIKE '%%latest'
 AND rules_mapping.mapping IS NULL
-AND (STR_TO_DATE(RIGHT(name, 14), "%%Y%%m%%d%%H%%i%%S") < NOW() - INTERVAL {nightly_age} DAY);
+AND (STR_TO_DATE(RIGHT(name, 14), "%%Y%%m%%d%%H%%i%%S") < NOW() - INTERVAL :nightly_age DAY);
 """
 
 
@@ -33,16 +34,17 @@ def cleanup_releases(trans, nightly_age, dryrun=True):
     # through two levels of Python string formatting. The first is here,
     # and the second happens at a low level of SQLAlchemy when the transaction
     # is being executed.
-    query = RELEASES_CLEANUP_CONDITION.format(nightly_age=nightly_age)
     if dryrun:
-        todelete = trans.execute("SELECT name FROM releases" + query).fetchall()
+        query = text("SELECT name FROM releases" + RELEASES_CLEANUP_CONDITION)
+        todelete = trans.execute(query, nightly_age=nightly_age).fetchall()
         print("Releases rows to be deleted:")
         if todelete:
             print("\n".join(itertools.chain(*todelete)))
         else:
             print("  - None")
     else:
-        trans.execute("DELETE releases FROM releases" + query)
+        query = text("DELETE releases FROM releases" + RELEASES_CLEANUP_CONDITION)
+        trans.execute(query, nightly_age=nightly_age)
 
 
 def cleanup_releases_json(trans, nightly_age, dryrun=True):
diff --git a/src/auslib/db.py b/src/auslib/db.py
index d2c04767..a696899f 100644
--- a/src/auslib/db.py
+++ b/src/auslib/db.py
@@ -211,10 +211,10 @@ class AUSTransaction(object):
         if not self.conn.closed:
             self.conn.close()
 
-    def execute(self, statement):
+    def execute(self, statement, **parameters):
         try:
             self.log.debug("Attempting to execute %s" % statement)
-            return self.conn.execute(statement)
+            return self.conn.execute(statement, **parameters)
         except Exception as exc:
             self.log.debug("Caught exception")
             # We want to raise our own Exception, so that errors are easily
```

Reply:
I think this is a good idea, but I'd prefer not to make a change to the database layer as part of this to avoid adding risk. I filed #3015 for this.

Reply:
Makes sense